### ISSUE TITLE
Implementing S

### ISSUE THREAD

**Issue by martinberlin** [martinberlin/udpx#9]
[Browsersync] Serving files from: examples
[Browsersync] Watching files...
17 00010001

Now the headers are like expected. 
@hputzek even if I tried to change the defaults here: https://github.com/hputzek/little-stream-protocol/compare/feature/header-bits?expand=1#diff-9c752b9294e4f2b8b64b038ac3ea0bf2L7
It keeps on sending other values. Please check it out if you want I will open an issue later.

It was not working since I was getting the order of the bits wrong:
```javascript
  let version = "000"
  let downstreamFlag = (options.header.downstreamFlag)?"1":"0"
  let compressedFlag = (options.header.compressedFlag)?"1":"0"
  //options.header.type.toString(2)
  let type = "001"
  let bin = version + downstreamFlag + compressedFlag + type
```
The exact bits that are documented should be respected or it won't generate the right byte (Note version is 3 bits, type is also 3 bits, rest being one)

Working testing branches:
https://github.com/martinberlin/udpx/tree/testing/s
https://github.com/hputzek/little-stream-protocol/tree/feature/header-bits

@IoTPanic I'm getting 3 extra bytes like I told you in skype chat. I think that's the reason the pixels are still not getting the brightness. 
Added some 55 brightness randomly to test:
![3littleBytes](https://user-images.githubusercontent.com/2692928/67034979-044ad400-f119-11e9-8416-fe6d202148da.png)

DEBUG Note that above here I'm sending it with the test tool that's why it looks random:
11 17 19 A 0 2D 0 0 50 0 0 A 0 17 17 18 1F 1C 24 1 26 11 12 D 1B E F 2 8 17 25 21 21 1C 16 B 24 16 D 11 27 24 19 5B 0 H  h.downstream = true
HEADER h.valid = true -> nodeId:23 session:25 frame:10 fragment:0
transactionLength: 45 transactionChecksum: 0 pyld: 1073556590 len: 37 m.valid: 1[ s ] Got valid message
Was a single packet, submitting to application instead of adding to buffer
Missing checkvalue
Clearing strand pyld length 45

2D 0 0 50 0 0 A 0 17 17 18 1F 1C 24 1 26 11 12 D 1B E F 2 8 17 25 21 21 1C 16 B 24 16 D 11 27 24 6B 73 75 6D 3A 20 30 20 

This 2D 0 0 should not come if I see that right, when I redirect the payload to pix.receive.

**Comment by martinberlin** [comment]
It's working. 
Just delete from .pio/libdeps the S library so PlatformIO pulls again. And test it with the branches in the last comment. Working good, seems that compression is also working, but cannot really say if it's ok or not. 
The total pixels seems a bit more in the Strip than what I'm sending, still didn't check why, but does not seem to be a problem for testing.

Snake modus is also there @hputzek and it's working non-compressed. I suspect there is some issue with the compression in either the sending part or receiving part, still need to debug it, but is not working with the snake. I though it was working before since I had the "random mode" on.
So I guess that will be the next steps, to make compression and then fragmentation work. To check it for real in a very large stripe. 
I have a 484 pix mini Strip on the wall, sent also one to Samuel and I have also some 5 meters RGB so let's try it together in the weekend. Will add the FPS counter later to measure at what speed we can receive. Thanks for the patience all... Was a hard one testing this!

**Comment by martinberlin** [comment]
@IoTPanic when you have some time, shall we deep debug and test how compression is working?
Then we could close #5 
Will like to do more tests over the weekend. 
And then we could follow with fragmentation. I would like to understand how this will happen, specially if what you are sending is compressed and fragmented at the same time. 
Great that started working. 
As a small bug I found out that it always makes "ghost" pixels at the end. So it's lighting up 2 or 3 leds more at the end, no matter what large you send from the tester, will find time to report on that with more details.

**Comment by hputzek** [comment]
> As a small bug I found out that it always makes "ghost" pixels at the end. So it's lighting up 2 or 3 leds more at the end

You're right, this is a problem of my pixels+s JS implementation I think - I will look into this

**Comment by hputzek** [comment]
https://github.com/hputzek/little-stream-protocol/issues/6

**Comment by IoTPanic** [comment]
> @IoTPanic when you have some time, shall we deep debug and test how compression is working?
> Then we could close #5
> Will like to do more tests over the weekend.
> And then we could follow with fragmentation. I would like to understand how this will happen, specially if what you are sending is compressed and fragmented at the same time.
> Great that started working.
> As a small bug I found out that it always makes "ghost" pixels at the end. So it's lighting up 2 or 3 leds more at the end, no matter what large you send from the tester, will find time to report on that with more details.

Yes, we can go over everything this weekend
